pyatv/mrp/__init__.py

```diff
@@ -359,16 +367,21 @@ def __init__(self, loop, protocol, remote):
             protobuf.DEVICE_INFO_UPDATE_MESSAGE)
 
     async def _get_current_power_state(self):
-        if self._power_state is None:
-            current_power_state = await self.update_power_state(self.protocol.device_info, protobuf.DEVICE_INFO_MESSAGE)
-            return current_power_state
+        if self._current_power_state is None:
+            updated_power_state = await self.update_power_state(self.protocol.device_info, protobuf.DEVICE_INFO_MESSAGE)
+            return updated_power_state
         
-        return self._power_state
+        return self._current_power_state
 
     @property
-    def power_state(self):
+    def listener(self):
```

Comment:
This is OK but you also need to provide a setter property, otherwise it's not possible to assign a listener. I assumed you took some inspiration from `PushUpdater`, so you can do it in the same way:

https://github.com/postlund/pyatv/blob/024284bca4565b3d75509fa82a509dcbeeeb7a96/pyatv/interface.py#L362